src/uwtools/resources/rocoto.jsonschema

```diff
@@ -1,70 +1,251 @@
+            "type": "object",
+            "properties": {
+              "attrs": {
+                "type": "object",
+                "properties": {
+                  "metatask": {
+                    "type": "string"
+                  },
+                  "cycle_offset": {
+                    "$ref": "#/$defs/TimePattern"
+                  },
+                  "state": {
                     "type": "string",
-                    "format": "uri"
+                    "enum": [
+                      "SUCCEEDED",
+                      "DEAD",
+                      "Succeeded",
+                      "Dead",
+                      "succeeded",
+                      "dead"
+                    ]
+                  },
+                  "threshold": {
+                    "type": "number",
+                    "minimum": 0,
+                    "maximum": 1
+                  }
                 },
-                "entities": {
-                    "type": "object"
-                }
+                "required": [
+                  "metatask"
+                ],
+                "additionalProperties": false
+              }
+            },
+            "additionalProperties": false
+          },
+          "^timedep(_.*)?$": {
+            "$ref": "#/$defs/TimePattern"
+          }
+        }
+      }
+    }
+  },
+  "type": "object",
+  "properties": {
+    "workflow": {
+      "type": "object",
+      "properties": {
+        "attrs": {
+          "type": "object",
+          "properties": {
+            "realtime": {
+              "type": "boolean"
+            },
+            "scheduler": {
+              "type": "string",
+              "enum": [
+                "sge",
+                "lsf",
+                "lsfcray",
+                "ll",
+                "torque",
+                "moabtorque",
+                "moab",
+                "pbspro",
+                "slurm",
+                "cobalt"
+              ]
+            },
+            "cyclethrottle": {
+              "type": "integer",
+              "minimum": 0
+            },
+            "taskthrottle": {
+              "type": "integer",
+              "minimum": 0
+            }
+          },
+          "required": [
+            "realtime",
+            "scheduler"
+          ]
+        },
+        "cycledefs": {
+          "type": "object",
+          "properties": {
+            "groupname": {
+              "type": "array",
+              "items": {
+                "$ref": "#/$defs/dateTimePattern"
+              }
             },
-            "required": ["attrs", "cycledefs", "log"]
+            "required": [
+              "groupname"
+            ]
+          }
+        },
+        "log": {
+          "type": "string",
+          "format": "uri"
+        },
+        "entities": {
+          "type": "object"
+        },
+        "tasks": {
+          "type": "object",
+          "$ref": "#/$defs/taskPattern"
+        },
+        "dependency": {
```

Comment:
This section will also need to go under the tasks sections and task definitions.

Reply:
Ah, sloppy pasting here from moving everything around, I'll move it over